bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java

```diff
@@ -0,0 +1,332 @@
+            pulseDown(id);
+            // try {
+            Thread.sleep(150);// sleep to not confuse the blinds
+            // } catch (InterruptedException e) {
+            // logger.error("EGate tiltDown error: {} ", e.toString());
+            // }
+        }
+    }
+
+    public void pulseUp(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=1;Priority=1;CheckNr=3415347;" + CR);
+    }
+
+    public void pulseDown(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=2;Priority=1;CheckNr=2764516;" + CR);
+    }
+
+    public void continuousUp(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=3;Priority=1;CheckNr=2867016;" + CR, 20000);
+    }
+
+    public void continuousDown(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=4;Priority=1;CheckNr=973898;" + CR, 20000);
+    }
+
+    public void stop(String id) {
+        sendCommand("Instruction=1;ID=" + id + ";Command=5;Priority=1;CheckNr=5408219;" + CR);
+    }
+
+    // @SuppressWarnings("null")
+    public void registerBlind(String id, ThingUID uid) {
+        logger.debug("Registring Blind id {} with thingUID {}", id, uid);
+        registeredBlinds.put(id, uid);
+    }
+
+    /**
+     * Send a command to the eGate Server.
+     */
+
+    private void sendCommand(String command) {
+        sendCommand(command, SOCKET_TIMEOUT_SEC);
+    }
+
+    @SuppressWarnings("null")
+    private synchronized void sendCommand(String command, int timeout) {
+        logger.debug("EGate got command: {}", command);
+
+        if (!isConnected()) {
+            logger.debug("no connection to Dominoswiss eGate server when trying to send command, returning...");
+            return;
+        }
+
+        // Send plain string to eGate Server,
+        try {
+            egateSocket.setSoTimeout(SOCKET_TIMEOUT_SEC);
+            writer.write(command);
+            writer.flush();
+        } catch (IOException e) {
+            logger.error("Error while sending command {} to Dominoswiss eGate Server {} ", command, e.toString());
+        }
+    }
+
+    private final Runnable pollingConfig = new Runnable() {
+        @Override
+        public void run() {
+            synchronized (lock) {
+                try {
+                    egateSocket = new Socket();
+                    egateSocket.connect(new InetSocketAddress(host, port));
+                    egateSocket.setSoTimeout(SOCKET_TIMEOUT_SEC);
+                    writer = new BufferedWriter(new OutputStreamWriter(egateSocket.getOutputStream()));
+                    writer.write("SilenceModeSet;Value=0;" + CR);
+                    writer.flush();
+
+                } catch (IOException e) {
+                    logger.debug("unknown socket host {}", host);
+                    try {
+                        egateSocket.close();
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED, e.toString());
```

Comment:
As this *is* the bridge, the thing status detail should be simply COMMUNICATION_ERROR.